scripts/pathfinders_validation.py

```diff
@@ -0,0 +1,559 @@
+                "Risk name": pa.Column(str, unique=True, report_duplicates="exclude_first"),
+                "Category": pa.Column(str, pa.Check.isin(PFEnums.RISK_CATEGORIES, error=PFErrors.isin)),
+                "Description": pa.Column(str, pa.Check.max_word_count(100, error=PFErrors.lt_100_words)),
+                "Likelihood score": pa.Column(str, pa.Check.isin(PFEnums.RISK_SCORES, error=PFErrors.isin)),
+                "Impact score": pa.Column(str, pa.Check.isin(PFEnums.RISK_SCORES, error=PFErrors.isin)),
+                "Mitigations": pa.Column(str, pa.Check.max_word_count(100, error=PFErrors.lt_100_words)),
+                "Risk Owner": pa.Column(str),
+            },
+            "checks": [pa.Check(lambda df: len(df.index) == 5, error="You must enter 5 risks.")],
+        },
+    },
+    "Underspend": {
+        "extract": {"id_tag": "PF-TABLE-ID-14", "worksheet_name": "Finances"},
+        "process": {"ignored_non_header_rows": [0]},
+        "validate": {
+            "columns": {
+                "What's your underspend for this financial year?": pa.Column(
+                    float, pa.Check.is_float(error=PFErrors.is_float)
+                ),
+            },
+        },
+    },
+    "Current underspend": {
+        "extract": {"id_tag": "PF-TABLE-ID-15", "worksheet_name": "Finances"},
+        "process": {"ignored_non_header_rows": [0]},
+        "validate": {
+            "columns": {
+                "What's your current underspend for this financial year?": pa.Column(
+                    float, pa.Check.is_float(error=PFErrors.is_float)
+                ),
+            },
+        },
+    },
+    "Underspend requested": {
+        "extract": {"id_tag": "PF-TABLE-ID-16", "worksheet_name": "Finances"},
+        "process": {"ignored_non_header_rows": [0]},
+        "validate": {
+            "columns": {
+                "How much underspend are you asking for with a credible plan?": pa.Column(
+                    float, pa.Check.is_float(error=PFErrors.is_float)
+                ),
+            },
+        },
+    },
+    "Spending plan": {
+        "extract": {"id_tag": "PF-TABLE-ID-17", "worksheet_name": "Finances"},
+        "process": {"merged_header_rows": [0], "ignored_non_header_rows": [0]},
+        "validate": {
+            "columns": {
+                "How do you plan to spend this value in the next financial year?": pa.Column(
+                    float, pa.Check.is_float(error=PFErrors.is_float)
+                ),
+            },
+        },
+    },
+    "Forecast spend": {
+        "extract": {"id_tag": "PF-TABLE-ID-18", "worksheet_name": "Finances"},
+        "process": {"ignored_non_header_rows": [0]},
+        "validate": {
+            "columns": {
+                "What is your forecast spend for the next financial year?": pa.Column(
+                    float, pa.Check.is_float(error=PFErrors.is_float)
+                ),
+            },
+        },
+    },
+    "Forecast and actual spend": {
```

Comment:
There is some issue with the cell references returned in the error messages for this table, so just adding this comment here as not sure where best to put it, as the cause needs investigating.

Basically though, at the moment we get told there are errors in cells B43 and B44, and we can see that these errors relate to the trailing whitespace in the `SPEND_TYPE` enum values. However, the corresponding values in the spreadsheet actually sit in cells B42 and B43, so it looks the cell reference calculation is failing here

Reply:
@cyrusdobbs could you confirm this is fixed? Not sure where relevant change lives and can't test as don't have updated spreadsheet with new tags - thanks

Reply:
Nice spot on this. There was a bug that caused the reference to be incorrect if any `ignored_header_rows` were not consecutive from 0. I've modified the code and made sure it's covered by tests now but I think it means that either: 
1. we keep `table.df` indexes as continuous from 0 but have another map like we do for column but from table idx to row idx
2. we don't reset the indexes and we keep the same pattern of adding the row index in question to the `table.first_row_idx`

2 means we avoid maintaining another mapping in `Table` and I don't think resetting the index gives us much benefit